### ISSUE TITLE
Activiti form customisation features/hooks in ADF 1/2

### ISSUE THREAD

**Issue by cijujoseph** [Alfresco/alfresco-ng2-components#1474]
**Type of issue:**  (check with "[x]")
```
- [x ] New feature request
- [ ] Bug  
- [ ] Support request
```

**New feature request:**
Activiti got some powerful form customisation features available within activiti-app UI. https://docs.alfresco.com/activiti/docs/dev-guide/1.5.0/#_start_and_task_form_customization.

Wondering if there are any hooks in ADF to achieve similar functions? If I look at the activiti-form component in ng2-activiti-form, the only @Output() event emitters are formSaved, formCompleted, formLoaded, onError and executeOutcome. 

Raising this issue to provide some samples, documentation, additional output event hook points etc so that users can do really powerful things in ADF

**Comment by magemello** [comment]
-    formRendered:function(form, scope) 

-    formFieldValueChanged:function(form, field, scope) 

-    formFieldFocus:function(form, field, scope) 

-    formFieldBlur:function(form, field, scope) 

-    formFieldPersonSelected:function(form, field, scope) 

-    formFieldPersonEmailSelected:function(form, field, scope) 

-    formFieldPersonRemoved:function(form, field, scope) 

-    formFieldGroupSelected:function(form, field, scope) 

-    formFieldGroupRemoved:function(form, field, scope) 

-    formFieldContentUploaded:function(form, field, scope) 

-    formFieldContentRemoved:function(form, field, scope) 

-    formFieldRestValuesSet:function(form, field, scope) 

-   formBeforeComplete:function(form, outcome, scope) 

-    formTableFieldValueChanged:function(form, field, columnDefinition, editRow, scope) 

-    formTableFieldFocus:function(form, field, columnDefinition, editRow, scope) 

-    formTableFieldBlur:function(form, field, columnDefinition, editRow, scope) 

-    formTableFieldRestValuesSet:function(form, field, columnDefinition, editRow, scope)

-	formTableRendered:function(form, field, columnDefinitions, editRow, scope) 

-	formTableBeforeComplete:function(form, field, editRow, scope) 

-	formTableBeforeCancel:function(form, field, editRow, scope) 

-	formValidateFieldValueChanged:function(form, field, scope) 

-	formValidateBeforeSubmit:function(form, outcome, scope) 

-	formTableValidateFieldValueChanged:function(form, field, columnDefinition, editRow, scope) 

-	formTableValidateBeforeComplete:function(form, field, editRow, scope) 

-   taskCompleted:function(taskId, form, scope) 

-    taskCompletedError:function(taskId, errorResponse, form, scope) 

-    taskSaved:function(taskId, form, scope) 

-    taskSavedError:function(taskId, errorResponse, form, scope)

**Comment by DenisVuyka** [comment]
Most of the existing events are based on the expectation of jQuery usage and html forms rather than components. I suggest we split that into iterations and provide only bare minimum that makes sense in the first wave. See #1528 PR for more details.

**Comment by magemello** [comment]
can you check in the list above which one are going to provide...and strike the one which doesn't make sense to implement?

**Comment by DenisVuyka** [comment]
Ok, but parameters and types will be different with our implementations, we are not giving `scope` and other angular-1 stuff.

**Comment by magemello** [comment]
The remaining events are going to be implemented as part of #1567
